File: BaekJoon Online Judge/C++/Step08-Math_1 Mathematical Thinking/2775.cpp
//
//  2775.cpp
//  Step08-Math_1 Mathematical Thinking
//
//  Created by Yebin Kim on 2019/10/23.
//  Copyright © 2019 김예빈. All rights reserved.
//

#include <iostream>

using namespace std;

// Time Complexity O(1)
int main() {
    cin.tie(NULL);
    ios::sync_with_stdio(false);
    
    int tc, k, n, i, j;
    int room[15][15];
    
    for(i = 1; i < 15; i++) {
        room[0][i] = i + 1;
        room[i][0] = 1;
    }
    
    for(i = 1; i < 15; i++)
        for(j = 1; j < 15; j++)
            room[i][j] = room[i][j - 1] + room[i - 1][j];
    
    cin >> tc;
    
    while(tc--) {
        cin >> k >> n;
        cout << room[k][n - 1] << "\n";
    }
    
    return 0;
}

File: BaekJoon Online Judge/Swift/etc/2775.swift
//
//  2775.swift
//  Step08-Math_1 Mathematical Thinking
//
//  Created by Yebin Kim on 2019/11/15.
//  Copyright © 2019 김예빈. All rights reserved.
//

import Foundation

let tc = Int(readLine()!)!

for _ in 0..<tc {
    let k = Int(readLine()!)!
    let n = Int(readLine()!)!
    
    var room = [[Int]]()
    
    for i in 0...k {
        var floor = [Int]()
        var sum = 0
        
        for j in 1...n {
            if i == 0 {
                floor.append(j)
            } else {
                sum += room[i - 1][j - 1]
                floor.append(sum)
            }
        }
        
        room.append(floor)
    }
    
    print(room.last!.last!)
}
